### ISSUE TITLE
dstport static field value can't be displayed in alerts description

### ISSUE THREAD

**Issue by Zenidd** [wazuh/wazuh#4352]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.10.2 | Analysisd | Manager | Packages&Sources | All |

Hi team, 

`dstport` field value isn't displayed when used inside a rule `<description>`:

### Steps to reproduce
**Example log**
```
date=2016-06-15 time=11:44:46 devname=Device_Name devid=FGTXXXX9999999999 logid=9999999999 type=utm subtype=webfilter eventtype=urlfilter level=warning vd="root" urlfilteridx=3 urlfilterlist="default" policyid=2 sessionid=1563645 user="" srcip=1.2.3.11 srcport=52414 srcintf="internal2" dstip=1.5.5.92 dstport=443 dstintf="wan2" proto=6 service=HTTPS hostname="4-edge-chat.facebook.com" profile="default" action=blocked reqtype=referral url="/p?partition=-2&cb=lz1k&failure=5&sticky_token=274&sticky_pool=atn2c06_chat-proxy" sentbyte=932 rcvdbyte=0 direction=outgoing msg="URL was blocked because it is in the URL filter list" crscore=30 crlevel=high
```

**Example rule**
```
  <rule id="111111" level="6">
    <if_sid>81620</if_sid>
    <description>The dstport value is '$(dstport)'</description>
  </rule>
```

**Analysisd alert**
```
** Alert 1576072312.353796: - local,syslog,sshd,
2019 Dec 11 14:51:52 nanoPill->/var/log/syslog
Rule: 111111 (level 6) -> 'The dstport value is '''
Src IP: 1.2.3.11
Src Port: 52414
Dst IP: 1.5.5.92
Dst Port: 443
date=2016-06-15 time=11:44:46 devname=Device_Name devid=FGTXXXX9999999999 logid=9999999999 type=utm subtype=webfilter eventtype=urlfilter level=warning vd="root" urlfilteridx=3 urlfilterlist="default" policyid=2 sessionid=1563645 user="" srcip=1.2.3.11 srcport=52414 srcintf="internal2" dstip=1.5.5.92 dstport=443 dstintf="wan2" proto=6 service=HTTPS hostname="4-edge-chat.facebook.com" profile="default" action=blocked reqtype=referral url="/p?partition=-2&cb=lz1k&failure=5&sticky_token=274&sticky_pool=atn2c06_chat-proxy" sentbyte=932 rcvdbyte=0 direction=outgoing msg="URL was blocked because it is in the URL filter list" crscore=30 crlevel=high
```

**ossec-logtest alert**
```
**Phase 2: Completed decoding.
       decoder: 'fortigate-firewall-v5'
       status: 'warning'
       srcuser: ''
       srcip: '1.2.3.11'
       srcport: '52414'
       dstip: '1.5.5.92'
       dstport: '443'
       url: '4-edge-chat.facebook.com'
       action: 'blocked'

**Phase 3: Completed filtering (rules).
       Rule id: '111111'
       Level: '6'
       Description: 'The dstport value is '''
**Alert to be generated.
```

As seen in the example above, dstport is missing in ossec-logtest description alerts either in analysisd sourced description alerts.

### The source of error

Wazuh [eventinfo.c](https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fwazuh%2Fwazuh%2Fblob%2Fv3.10.2%2Fsrc%2Fanalysisd%2Feventinfo.c&sa=D&sntz=1&usg=AFQjCNH32W7tJZINv5vPtY9JKD1USmTjSA) file contains the parsing mechanism for extracted fields to be used as output in the rules description. The checking for `dstport` is missing:
https://github.com/wazuh/wazuh/blob/7aa4d46661f8b0f97b0969634482626f30866731/src/analysisd/eventinfo.c#L1086-L1116

Adding the check for `dstport` to the static fields checks should be enough.

Greetings, JP Sáez

**Comment by jnasselle** [comment]
This behavior was reported in #7044 and fixed as part of `wazuh-logtest` in #7073.

Regards,
Nico

**Comment by yanazaeva** [comment]
Hi team,

I am going to reopen this issue since this field is not working in v4.1.5. 

One of our community users sent me his decoders and rules, thanks to which we discovered that this field is not working. I will leave them below if you want to check. 

**Decoders:**
```
<decoder name="cisco_fmc">
    <type>syslog</type>
    <prematch>FirepowerMC</prematch>
</decoder>
<decoder name="cisco_fmc-fields">
    <parent>cisco_fmc</parent>
    <regex offset="after_parent">[(\d+:\d+:\d+)] (\D+) [Impact: (\S+)] From "(\S+)" at (\D+\s+\d+\s+\d+:\d+:\d+\s+\d+) UTC [Classification: (\.+)] [Priority: (\d)] {(\.+)} (\d+.\d+.\d+.\d+):(\d+) \((\.+)\)->(\d+.\d+.\d+.\d+):(\d+) \((\.+)\)</regex>
    <order>snort.ruleid, snort.rulecategory, impact, fpip, datetime, classification, priority, protocol, srcip, srcport, srccountry, dstip, dstport, dstcountry</order>
</decoder>
```

**Rules:**
```
    <rule id="202200" level="12">
        <decoded_as>cisco_fmc</decoded_as>
        <regex>[\d+:\d+:\d+]</regex>
        <description>Cisco FMC: Alert $(snort.rulecategory) - $(classification) $(srcip):$(srcport) $(srccountry) -> $(dstip):$(dstport) $(dstcountry) </description>
    </rule>
```

**Log:**
`2021 Apr 27 14:36:44 firepower.best.host->111.111.111.111 Apr 29 08:42:05 firepower.best.host FirepowerMC: [1:43687:2] "INDICATOR-COMPROMISE Suspicious .top dns query" [Impact: Unknown] From "123.123.123.123" at Thu Apr 29 08:42:03 2021 UTC [Classification: Misc Activity] [Priority: 3] {udp} 123.123.123.123:123 (unknown)->321.321.321.321:321 (Morroko)`

According to the rule, the destination port should be displayed before the destination country, which in this case is Morroko.

When you run _ossec-logtest_ this is the output: 
![image](https://user-images.githubusercontent.com/67015786/117136671-fa46ff80-ada8-11eb-9089-80a55adec628.png)

For _wazuh-logtest_ we obtain the same behavior:
![image](https://user-images.githubusercontent.com/67015786/117136761-1ba7eb80-ada9-11eb-82c5-a69ed8d0c716.png)

When the event is being processed by _Analysisd_ we obtain the same behavior too: 
![image](https://user-images.githubusercontent.com/67015786/117136875-4003c800-ada9-11eb-8f2f-57435b048247.png)

However, if we change the field name (I've changed it to _dstport1_), the behavior is the expected one: 
![image](https://user-images.githubusercontent.com/67015786/117137016-6fb2d000-ada9-11eb-98ce-1fcc8f8ca988.png)

Let me know if you need anything else.

Regards,
Yana.

**Comment by juliancnn** [comment]
I will close the issue as this bug is known in 4.1 and this fix goes into 4.2.
Regards,
Julian.
